modules/service/src/main/scala/lucuma/odb/service/ProgramUserService.scala

```diff
@@ -161,6 +167,25 @@ object ProgramUserService:
           _  <- ResultT(updateProperties(set, sql"$program_user_id"(id)))
         yield id).value
 
+      override def changeProgramUserRole(
+        input: ChangeProgramUserRoleInput
+      )(using Transaction[F]): F[Result[ProgramUser.Id]] =
+        session
+          .prepare(Statements.SelectLinkData)
+          .flatMap(_.option(input.programUserId))
+          .flatMap:
+            case None                     =>
+              OdbError.InvalidProgramUser(input.programUserId, s"ProgramUser ${input.programUserId} was not found.".some).asFailureF
+            case Some((pid, prevRole, _)) =>
```

Comment:
Would it be worth the trouble to compare the previous and new roles and return a no-op warning if they are the same?